components/script/dom/bindings/root.rs

```diff
@@ -274,7 +274,7 @@ impl RootCollection {
     unsafe fn unroot(&self, object: *const dyn JSTraceable) {
         assert_in_script();
         let roots = &mut *self.roots.get();
-        match roots.iter().rposition(|r| std::ptr::eq(*r, object)) {
+        match roots.iter().rposition(|r| std::ptr::eq(*r as *const (), object as *const ())) {
```

Comment:
Perhaps we can add a FIXME comment about the caveat with using `ptr::eq` for wide-pointer comparison and switching to addr_eq.

Reply:
Yes, good idea.